### PR TITLE
initial edits for quickstart, and slight edits in examples ReadMes

### PR DESCRIPTION
**Pull request by christinerose**
Addressed some concerns from [feedback](https://github.com/ocaml/dune/issues/5915), but there are still more to go. There are still some unanswered questions for which I'm awaiting answers. I will keep updating this file as I go, but I wanted to start a PR in case what I've done so far isn't accurate or needs more clarification. 

Signed-off-by: Christine Rose <professor.rose@gmail.com>

### CODE REVIEW

**Comment by christinerose** [comment]
I pushed this last commit because it's been sitting on my desktop now for a few weeks! I've been pulled away on other things, but I hope to get back to this properly over the break.

File: doc/quick-start.rst [hunk 1]
```diff
@@ -91,11 +110,19 @@ which you can dive deeper into Dune's capabilities:
   ``dune-project`` file whenever you build your project. You shouldn't need to
   worry about this, but you can see :ref:`opam-generation` for details.
 * The ``dune`` files in each directory specify the component to be built with
-  the files in that directory. For details on dune files, see :ref:`dune-files`.
+  the files in that directory. For details on ``dune`` files, see :ref:`dune-files`.
 
 Initializing a Library
 ----------------------
 
+There are two kinds of libraries: a shared library and a static library. A shared
```
Comment (Alizter) [inline doc/quick-start.rst:118]: The descriptions here seem a bit muddled. This is also the first time I have heard of shared vs static libraries.

File: doc/quick-start.rst [hunk 2]
```diff
@@ -169,16 +197,22 @@ Next, build your new program in a shell using this command:
...
 (including non-Windows systems). 
 
 Finally, run it with the following command to see that it worked. In 
 fact, the executable can both be built and run in a single
-step with ``dune exec ./hello_world.exe``.
+step:
+
+.. code:: bash
+
+    dune exec ./hello_world.exe
```
[the remark this answers is not in the document]
Comment (christinerose) [reply doc/quick-start.rst:211]: Thank you so much for all your comments and suggestions, Alizter!

File: doc/quick-start.rst (hunk 1 above)
Comment (christinerose) [reply doc/quick-start.rst]: I removed it.

**Pull request merged**